microbench/src/main/java/alluxio/fsmaster/FileSystemBench.java

```diff
@@ -0,0 +1,208 @@
+import com.google.common.util.concurrent.MoreExecutors;
+import org.apache.hadoop.fs.Path;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Assert;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.infra.ThreadParams;
+import org.openjdk.jmh.results.format.ResultFormatType;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.CommandLineOptions;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+import java.io.IOException;
+import java.util.concurrent.Semaphore;
+
+@SuppressFBWarnings("UWF_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR")
+public class FileSystemBench {
+  @State(Scope.Benchmark)
+  public static class FileSystem {
+    /**
+     * Use standalone parameters if you've started a standalone server using the main function in
+     * {@link FileSystemBase}.
+     */
+    @Param({"localhost"})
+    String mServerIpAddress;
+    @Param({"0"})
+    int mServerPort;
+
+    @Param({ "ALLUXIO_GRPC_SERVER", "BASIC_GRPC_SERVER" })
+    public FileSystemBase.ServerType mServerType;
+    @Param({"2"})
+    public int mNumGrpcChannels;
+    @Param({"100"})
+    public int mNumConcurrentCalls;
+
+    FileSystemBase mBase = new FileSystemBase();
+
+    @Setup(Level.Trial)
+    public void setup() throws Exception {
+      Assert.assertTrue("if standalone server address is specified, host must be specified and "
+              + "the port must be greater than 0",
+          mServerType != FileSystemBase.ServerType.STANDALONE
+              || (!mServerIpAddress.isEmpty() && mServerPort > 0));
+      mBase.init(mServerType, mNumGrpcChannels, mServerIpAddress, mServerPort);
+    }
+
+    @TearDown
+    public void tearDown() {
+      mBase.tearDown();
+    }
+  }
+
+  @State(Scope.Thread)
+  public static class ThreadState {
+    public final AlluxioURI mURI = new AlluxioURI("/");
+    public final GetStatusPOptions mOpts = GetStatusPOptions.getDefaultInstance();
+    Semaphore mSemaphore;
+
+    @Setup(Level.Trial)
+    public void setup(FileSystem fs) {
+      mSemaphore = new Semaphore(fs.mNumConcurrentCalls);
+    }
+  }
+
+  @State(Scope.Thread)
+  public static class AlluxioBenchThreadState extends ThreadState {
+    alluxio.client.file.FileSystem mFs;
+
+    @Setup(Level.Trial)
+    public void setup() {
+      mFs = alluxio.client.file.FileSystem.Factory.create(Configuration.global());
+    }
+
+    @TearDown
+    public void tearDown() throws IOException {
+      mFs.close();
+    }
+  }
+
+  @Benchmark
+  public void alluxioGetStatusBench(Blackhole bh, AlluxioBenchThreadState ts) throws Exception {
+    ts.mSemaphore.acquire();
```

Comment:
actually can you only use the semaphore in the async bench?

Reply:
done.